internal/provider/aws/provider.go

```diff
@@ -857,31 +857,86 @@ func (p *awsProvider) ReconcileExternalIP(
 	instanceID string,
 	externalIP *v1alpha1.ExternalIP,
 ) (v1alpha1.ExternalIPStatus, error) {
-	status := externalIP.Status
+	status := v1alpha1.ExternalIPStatus{
+		State:      v1alpha1.ExternalIPStatePending,
+		Conditions: []kmetav1.Condition{},
```

Comment:
I think you should not wipe the conditions (Deepcopy).

Reply:
I don't understand no condition is wiped the update is done in another function and the externalIP status is deepcopied
Am I missing something ?

Reply:
Yes, you should keep conditions, at least to keep the lastTransitionTime:
```suggestion
		Conditions: slices.Clone(externalIP.Status.Conditions),
```